src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/SymbolicRegexMatcher.cs

```diff
@@ -164,8 +164,11 @@ private SymbolicRegexMatcher(SymbolicRegexNode<TSet> rootNode, int captureCount,
                 _fixedMatchLength = findOptimizations.MinRequiredLength;
             }
 
-            if (findOptimizations.FindMode != FindNextStartingPositionMode.NoSearch &&
-                findOptimizations.LeadingAnchor == 0) // If there are any anchors, we're better off letting the DFA quickly do its job of determining whether there's a match.
+            // Store the find optimizations that can be used to jump ahead to the next possible starting location.
+            // If there's a leading beginning anchor, the find optimizations are unnecessary on top of the DFA's
+            // handling for beginning anchors.
+            if (findOptimizations.IsUseful &&
+                findOptimizations.LeadingAnchor is not RegexNodeKind.Beginning)
```

Comment:
NIT: should this second check for LeadingAnchor being Beginning be also collapsed to the IsUseful method since this is the only place it gets used?

Reply:
I'd prefer not to, purely as a layering thing.  It's true that right now nothing else calls this IsUseful, but that property is encoding RegexFindOptimization's details about whether it provides any useful optimizations... then at the NonBacktracking layer, NonBacktracking is special-casing something it knows it can do better for one particular optimization provided.